### DISCUSSION TITLE
Building a pagination element manually

### DISCUSSION THREAD

**Discussion by ttelik** [Budibase/budibase#8421] [Help]
Since pagination is not available in data providers where the provider is not a database (in my case it's based on a relationship between database entries), is there a way to achieve this manually? Essentially I'm looking to set the limit of the provider's results to 1 and then somehow adjust the index using a state variable or other means.

Even if pagination were available, I am looking to have customized buttons for previous/next.

**Comment by melohagan** [comment]
Hey @ttelik 

Does this help? https://github.com/Budibase/budibase/discussions/7313#discussioncomment-3430353

**Reply by ttelik** [reply]
That's certainly interesting, but I was hoping to do this in budibase db. Perhaps I'll move the databases over to SQL if that's the only way to implement it.
